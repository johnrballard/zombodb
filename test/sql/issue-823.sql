CREATE TABLE issue823
(
    id         SERIAL8 NOT NULL PRIMARY KEY,
    value_term text    NOT NULL
);
CREATE INDEX idxissue823
    ON issue823 USING zombodb ((issue823.*));

insert into issue823(value_term)
values ('Bellows Under the bridge');
insert into issue823(value_term)
values ('Burger Meister');
insert into issue823(value_term)
values ('Chemical Smells From Somewhere');
insert into issue823(value_term)
values ('Chowderhead Luncheon');
insert into issue823(value_term)
values ('Green Olive Oyl');
insert into issue823(value_term)
values ('Left Fenders');
insert into issue823(value_term)
values ('Wiper Blade Sandwich Mountain');

SELECT zdb.define_analyzer('case_sensitive_phrase','{"type":"custom", "tokenizer":"standard", "filter":["zdb_truncate_to_fit"]}');

SELECT zdb.define_field_mapping('issue823'::regclass, 'value_term',
                                '{"type": "text", "copy_to": "zdb_all", "analyzer": "phrase", "fielddata": true, "fields":{"exact_case":{"type":"text", "analyzer":"case_sensitive_phrase", "fielddata":true}}}');

REINDEX INDEX idxissue823;

select * from zdb.dump_query('issue823', 'value_term.exact_case:the');
SELECT * FROM issue823 WHERE issue823 ==> 'value_term.exact_case:the';
SELECT upper(term) term, count, term AS exact_term from zdb.tally('issue823'::regclass, 'value_term.exact_case', 'FALSE', '^[cC].*', ''::zdbquery, 5000, 'term'::termsorderby);

DROP TABLE issue823;