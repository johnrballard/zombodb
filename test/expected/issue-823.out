CREATE TABLE issue823
(
    id         SERIAL8 NOT NULL PRIMARY KEY,
    value_term text    NOT NULL
);
CREATE INDEX idxissue823
    ON issue823 USING zombodb ((issue823.*));
insert into issue823(value_term)
values ('Bellows Under the bridge');
insert into issue823(value_term)
values ('Burger Meister');
insert into issue823(value_term)
values ('Chemical Smells From Somewhere');
insert into issue823(value_term)
values ('Chowderhead Luncheon');
insert into issue823(value_term)
values ('Green Olive Oyl');
insert into issue823(value_term)
values ('Left Fenders');
insert into issue823(value_term)
values ('Wiper Blade Sandwich Mountain');
SELECT zdb.define_analyzer('case_sensitive_phrase','{"type":"custom", "tokenizer":"standard", "filter":["zdb_truncate_to_fit"]}');
 define_analyzer 
-----------------
 
(1 row)

SELECT zdb.define_field_mapping('issue823'::regclass, 'value_term',
                                '{"type": "text", "copy_to": "zdb_all", "analyzer": "phrase", "fielddata": true, "fields":{"exact_case":{"type":"text", "analyzer":"case_sensitive_phrase", "fielddata":true}}}');
 define_field_mapping 
----------------------
 
(1 row)

REINDEX INDEX idxissue823;
select * from zdb.dump_query('issue823', 'value_term.exact_case:the');
           dump_query           
--------------------------------
 {                             +
   "match": {                  +
     "value_term.exact_case": {+
       "query": "the",         +
       "boost": 1.0            +
     }                         +
   }                           +
 }
(1 row)

SELECT * FROM issue823 WHERE issue823 ==> 'value_term.exact_case:the';
 id |        value_term        
----+--------------------------
  1 | Bellows Under the bridge
(1 row)

SELECT upper(term) term, count, term AS exact_term from zdb.tally('issue823'::regclass, 'value_term.exact_case', 'FALSE', '^[cC].*', ''::zdbquery, 5000, 'term'::termsorderby);
    term     | count | exact_term  
-------------+-------+-------------
 CHEMICAL    |     1 | Chemical
 CHOWDERHEAD |     1 | Chowderhead
(2 rows)

DROP TABLE issue823;
